doc.go

```diff
@@ -0,0 +1,142 @@
+/*
+Package thanos is a set of components that
+can provide highly available Prometheus
+setup with long term storage capabilities.
+
+See https://github.com/improbable-eng/thanos/blob/master/docs/getting_started.md for getting start.
+
+Configuration watches and reloads
```

Comment:
This definitely should go to `pkg/reloader package instead! And below to corresponded packages.  But examples makes total sense (: Just lest move them to proper places.

Also can we use some collapsable examples? This is quite neat! https://godoc.org/github.com/fluhus/godoc-tricks#Examples